subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/MultipleOriginProcessor.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.compile.processing;
+
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.Processor;
+
+public class MultipleOriginProcessor extends IncrementalProcessor {
```

Comment:
What about something like `ValidatingMultipleOriginProcessor`, to recognize the fact it doesn't do anything by itself, but performs validation?

Reply:
I wanted to keep it somewhat short, otherwise I'd have to call this `MultipleOriginValidatingProcessorDecorator` :) I'll add some JavaDoc about its purpose though.